olive/model.py

```diff
@@ -49,6 +49,7 @@ def __init_subclass__(cls, **kwargs) -> None:
     def __init__(
         self,
         framework: Framework,
+        model_file_format: ModelFileFormat,
```

Comment:
Nit: Please also complete the docs to introduce the model file format. (it is also fine to be added with metadata changes with some separated PR.